catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java

```diff
@@ -84,19 +88,15 @@ private Object executeRemoveFromStore() throws Exception {
         CatalogFacade catalogProvider = getCatalog();
 
         if (hasFilter()) {
-            QueryImpl query = new QueryImpl(getFilter());
-
-            query.setRequestsTotalResultsCount(true);
-            query.setPageSize(-1);
-
-            Map<String, Serializable> properties = new HashMap<>();
-            properties.put("mode", "native");
-
-            SourceResponse queryResponse = catalogProvider.query(new QueryRequestImpl(query,
-                    properties));
-
-            final List<String> idsFromFilteredQuery = queryResponse.getResults()
-                    .stream()
+            ResultIterable resultIterable = new ResultIterable(catalogFramework,
+                    new QueryRequestImpl(new QueryImpl(getFilter(),
+                            1,
+                            PAGE_SIZE,
+                            SortBy.NATURAL_ORDER,
+                            true,
+                            TimeUnit.SECONDS.toMillis(30)), false));
+
+            final List<String> idsFromFilteredQuery = resultIterable.stream()
                     .map(result -> result.getMetacard()
                             .getId())
                     .collect(Collectors.toList());
```

Comment:
❗️ The `Iterable` will loop over _every_ metacard that matches the filter and will load them all in memory. The same thing would have happened in the past since we were asking for all (`-1`) metacards, but now that we're capping the number, no more than 1,000 metacards will be returned.

What we need to do here is either use the `ResultListIterable` to process things in batches, or keep using the `QueryResponse` in a loop until the number of metacards returned is 0. You can look at this PR for an example of how this can be done: https://github.com/codice/ddf/pull/2204.